include/ignition/common/Util.hh

```diff
@@ -160,6 +161,12 @@ namespace ignition
     void IGNITION_COMMON_VISIBLE addFindFileURICallback(
         std::function<std::string(const URI &)> _cb);
 
+    /// \brief Get a pointer to the global system paths that's used by all
+    /// the findFile functions.
+    /// Caller should not assume ownership.
```

Comment:
Are there any other warnings that should go along with this?  Does this mean that the caller can mutate the global system paths and impact other users?

Reply:
The idea is to give the caller the ability to do that. The impact will depend on who's using the global system paths though. The `MeshManager` happens to use it. Whoever calls `common::findFile` will use it.But there are many places where custom `SystemPaths` objects are used, such as when `ign-gazebo` loads worlds,

Reply:
I agree, that's a bit difficult to document.  I think just enough to make sure that if people start messing with paths orders that they would have a place to look. Maybe something like:

```
The returned instance has global shared state for a given process.  Care should be taken when manipulating global system paths.
```

Reply:
Thanks, added it here: b4aa1bce44de23b63d54b3186407eb9d81ab4c4d